veps/sig-compute/58-amd-sev-snp-enablement/vep.md

```diff
@@ -0,0 +1,161 @@
+-->
+KubeVirt supports SEV and SEV-ES for memory encryption, but lacks SEV-SNP enahnced memory integrity proctection. Organizations are now asking for workloads to include the strongest confidential computing secruity guarantees. 
+
+## Goals
+
+<!--
+The desired outcomeo
+-->
+- **Enable SEV-SNP VM Deployment**: Provide users the ability to deploy VMs that are utilizing AMD SEV-SNP technology. 
+- **Extend the existing SEV Feature**: Currently the SEV sits behind the `WorkloadEncryptionSEV` adding SEV-SNP to feature is the next logical step. 
+- **Extend Existing SEV Infrastructure**: Use the existing SEV node labeling, schedulin, and domain generation infrastructure.
+
+## Non Goals
+
+<!--
+Why this enhancement is important Limitations to the scope of the design
+-->
+- **Intel TDX Support**: This proposal is targeted only to AMD SEV and enabling Intel TDX support shold targeted in a seperate VEP.
+- **Custom Attestation Services**: Only focus on the enabling the capabilities through the standard interfaces and not building a custom attestation infrastructure
+- **SEV migration**: No automatic upgrade path from SEV/SEV-ES to SEV-SNP, this will be required to done by explicit user configuration.
+
+
+## Definition of Users
+
+<!--
+Who is this feature set intended for
+-->
+- **Cluster Admins**: Responsible for enabling and managing SEV-SNP capabilities in Kubernetes Clusters.
+- **Developers**: Deploys confidential workloads that require SEV-SNP.
+
+## User Stories
+
+<!--
+List of user stories this design aims to solve
+-->
+
+- As a Cluster Admin, I want to enable SEV-SNP on cluster nodes so that tenants can deploy Confidential Compute workloads with SEV-SNP enabled.
+- As a Cluster Admin, I need to ensure that my clusters have the correct labeling that shows the correct SEV and SEV-SNP labels are showing up on my nodes.
+
+- As a Developer, I want to deploy VMs on a platform with SEV-SNP protection to meet compliance requirements for memory integrity.
+- As a Developer, I want to deploy VMs without needing to understand the without understanding the low level AMD technology.
+
+
+## Repos
+
+<!--
+List of repose this design impacts
+-->
+https://github.com/kubevirt/kubevirt
+
+## Design
+
+<!--
+This should be brief and concise. We want just enough to get the point across
+-->
+The SEV-SNP feature extend the exisitn SEV features by adding a `secureNestedPaging` boolean flag to the current Launch Security API. When enabled the libvirt XML will contain `<launcSecurity type='sev-snp'>` using the default QEMU policy. the implementation will use existing SEV feature gate and extend the node labeling, by labeling nodes with `sev-snp.node.kubevirt.io`. The node labeller will detect SEV-SNP capabilities from libvirt domain capabilities then apply the label on the node, while the node selector renderer will be extended to include SEV-SNP scheudling. The feature will alo generate the domain XML to contain the required fields like `cbitpos`. 
```

Comment:
Can you provide an example of the generated libvirt domain XML below in API examples just so folks can understand how we end up configuring this?